buildconfig/class_maker.py

```diff
@@ -78,13 +79,13 @@ def write_header(subproject, classname, filename, args):
   File change history is stored at: <https://github.com/mantidproject/mantid>
   Code Documentation is available at: <http://doxygen.mantidproject.org>
 */
-class MANTID_%s_DLL %s%s {
-public:%s};
+class MANTID_{}_DLL {}{} {{
+public:{}}};
 
-} // namespace %s
-} // namespace Mantid
+}} // namespace {}
+}} // namespace Mantid
 
-#endif /* %s */""" % (guard, guard, subproject,
+#endif /* {} */""".format(guard, guard, subproject,
```

Comment:
Here one could save a bit of repetition with regards to the format parameters if named arguments were used:
```python
"""#ifndef {guard}
#define {guard}
#include "Mantid{subproject}/Dll.Config.h"
# [...]
#endif /* {guard} */""".format(guard = guard, subproject = subproject)
```
 Might also make the template strings easier to comprehend.